lib/types.nix

```diff
@@ -791,6 +797,46 @@ rec {
         };
       };
 
+    configuration = { class ? null, allowUnknownClass ? false, allowLegacyType ? false }:
```

Comment:
This feels out of scope for this PR, it doesn't appear needed for `class` to work, I'd just remove this

Reply:
If we merge this simultaneously, we can use `types?configuration` as a simple test whether classes are supported.
I agree that it is technically separable though. Is my reason for inclusion good enough?

Reply:
I don't think so :sweat_smile:. We can already check for classes support now with `functionArgs evalModules ? class` (though maybe we want more clear versioning scheme in the future, maybe `lib.modules.interfaceVersion` or so.).